### PR TITLE
Add possibility to scale observable parameters with dataset integral

### PR DESCRIPTION
**Pull request by vonsturm**
Needs testing and read ranges method can also be used when reading fit ranges.

### CODE REVIEW

**Comment by vonsturm** [comment]
Correctly scale parameters and reset names `p0 -> 0, p1 -> 1` etc.

**Comment by gipert** [comment]
Can you add a test config for this? You can copy from the `test` directory

**Comment by vonsturm** [comment]
I don't get it... Why do the tests fail with the last commit??? No sense...

**Comment by vonsturm** [comment]
Test-config will add...

**Comment by gipert** [comment]
(all tests will fail until #3 is merged)

**Comment by vonsturm** [comment]
Ok please review. From my side this is good to go.

**Comment by vonsturm** [comment]
PS: As I said the only thing that can still be changed is the reuse the `CheckAndStoreRanges` method

File: src/GerdaFitter.cc [hunk 1]
```diff
@@ -1124,3 +1189,82 @@ TF1 GerdaFitter::ParseTFormula(std::string prefix, std::string expr, double rang
...
+
+double GerdaFitter::IntegrateHistogram2D(
+    TH2* /*h*/,
+    std::vector<std::pair<double,double>> /*x_range*/,
+    std::vector<std::pair<double,double>> /*y_range*/
+)
+{ 
+    BCLog::OutSummary("Implement me : IntegrateHistogram::TH2");
+    return 0;
+}
```
Comment (gipert) [inline src/GerdaFitter.cc:1271]: Queste non possono essere eliminate?

File: src/GerdaFitter.cc [hunk 2]
```diff
@@ -540,6 +485,126 @@ GerdaFitter::GerdaFitter(json outconfig) : config(outconfig) {
...
+                }
+                // throw exception if parameter is undefined
+                if (!exists) throw std::runtime_error(
+                    "fit parameter '" + parname + "' not found, is it defined in \"parameters\"?"
+                );
+            }
+
+            // if requested substitute all parameters with :
+            // <integral-range> X <parameter> in formula
+            if (el.value().contains("scale-with-pdf-integral")) {
```
Comment (gipert) [inline src/GerdaFitter.cc]: `multiply-by-pdf-integral` would perhaps be more clear, `scale` is more generic

File: src/GerdaFitter.cc (hunk 1 above)
Comment (vonsturm) [reply src/GerdaFitter.cc:1271]: IntegrateHistogram1D viene usato quindi no. IntegrateHistogram2D ho messo se in futuro vuoi usarlo.

File: src/GerdaFitter.cc (hunk 2 above)
Comment (vonsturm) [reply src/GerdaFitter.cc]: ... ne abbiamo parlato e ho messo i nomi che avevamo deciso, se non ti piace cambialo

**Comment by gipert** [comment]
Perche' fai queste sottrazioni qui?

https://github.com/gipert/gerda-fitter/pull/5/files#diff-0c4423d154a02bfb6ac9fc6d6a2b0b35R963-R964

**Comment by vonsturm** [comment]
> Perche' fai queste sottrazioni qui?
> 
> https://github.com/gipert/gerda-fitter/pull/5/files#diff-0c4423d154a02bfb6ac9fc6d6a2b0b35R963-R964

Infatti la puoi togliere che e' piu' chiaro. Ci avevo gia pensato.

**Review by gipert — APPROVED** [review]

**Pull request merged**